exercises/hamming/.meta/src/reference/java/Hamming.java

```diff
@@ -1,36 +1,26 @@
+import java.util.function.IntPredicate;
+import java.util.stream.IntStream;
+
 class Hamming {
-    private final int hammingDistance;
+    private final String leftStrand, rightStrand;
 
     Hamming(String leftStrand, String rightStrand) {
-        hammingDistance = computeHammingDistance(leftStrand, rightStrand);
-    }
-
-    int getHammingDistance() {
-        return hammingDistance;
-    }
-
-    private int computeHammingDistance(String leftStrand, String rightStrand) {
-        if (leftStrand.length() == 0 && rightStrand.length() == 0) {
-            return 0;
-        } else if (leftStrand.length() == 0) {
-            throw new IllegalArgumentException("left strand must not be empty.");
-        } else if (rightStrand.length() == 0) {
-            throw new IllegalArgumentException("right strand must not be empty.");
-        } else if (leftStrand.length() != rightStrand.length()) {
-            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
+        String exceptionMessage = "leftStrand and rightStrand must be of equal length.";
+        if (leftStrand.length() != rightStrand.length()) {
+            if (leftStrand.isEmpty()) {
+                exceptionMessage = "left strand must not be empty.";
+            }
+            if (rightStrand.isEmpty()) {
+                exceptionMessage = "right strand must not be empty.";
+            }
+            throw new IllegalArgumentException(exceptionMessage);
         }
-
-        int length = leftStrand.length();
-        int distance = 0;
-        for (int i = 0; i < length; i++) {
-            distance += hammingContributionAt(i, leftStrand, rightStrand);
-        }
-
-        return distance;
+        this.leftStrand = leftStrand;
+        this.rightStrand = rightStrand;
     }
 
-    private int hammingContributionAt(int index, String leftStrand, String rightStrand) {
-        return leftStrand.charAt(index) != rightStrand.charAt(index) ? 1 : 0;
+    long getHammingDistance() {
```

Comment:
Any reason why you've changed this to `long`? Now it won't match the starter implementation 🙂

Reply:
@FridaTveit, you're right, I should have modified the starter implementation. When I think of it, maybe it's not worth to update the exercise for this matter. I'll add the `int` cast in the return statement!

Reply:
Great, thanks! 🙂